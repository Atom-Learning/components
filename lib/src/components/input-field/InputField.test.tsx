import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { axe } from 'jest-axe'
import * as React from 'react'

import { Form } from '..'
import { InputField } from '.'

describe(`InputField component`, () => {
  it('renders a field with a text input', async () => {
    const { container } = render(
      <Form>
        <InputField
          label="INPUT FIELD"
          name="INPUT FIELD"
          css={{ m: 'auto', height: 100, width: 100 }}
          placeholder="INPUT FIELD"
        />
      </Form>
    )

    await screen.getByPlaceholderText('INPUT FIELD')

    expect(container).toMatchSnapshot()
  })

  it('renders a field with a text input - has no programmatically detectable a11y issues', async () => {
    const { container } = render(
      <Form>
        <InputField
          label="INPUT FIELD"
          name="INPUT FIELD"
          placeholder="INPUT FIELD"
        />
      </Form>
    )

    expect(await axe(container)).toHaveNoViolations()
  })

  it('renders a field with a number input', async () => {
    const { container, findByPlaceholderText } = render(
      <Form>
        <InputField
          label="INPUT FIELD"
          name="INPUT FIELD"
          css={{ m: 'auto', height: 100, width: 100 }}
          type="number"
          placeholder="001"
        />
      </Form>
    )

    const input = await findByPlaceholderText('001')

    expect(input).toHaveAttribute('inputmode', 'numeric')
    expect(input).toHaveAttribute('pattern', '[0-9]*')
    expect(container).toMatchSnapshot()
  })

  it('renders a field with a number input - has no programmatically detectable a11y issues', async () => {
    const { container } = render(
      <Form>
        <InputField
          label="INPUT FIELD"
          name="INPUT FIELD"
          type="number"
          placeholder="001"
        />
      </Form>
    )

    expect(await axe(container)).toHaveNoViolations()
  })

  it('renders a field with a disabled input', async () => {
    const { findByPlaceholderText } = render(
      <Form>
        <InputField
          label="INPUT FIELD"
          name="INPUT FIELD"
          css={{ m: 'auto', height: 100, width: 100 }}
          placeholder="INPUT FIELD"
          disabled
        />
      </Form>
    )

    const input = await findByPlaceholderText('INPUT FIELD')

    expect(input).toHaveAttribute('disabled')
  })

  it('renders a field with a disabled input - has no programmatically detectable a11y issues', async () => {
    const { container } = render(
      <Form>
        <InputField
          label="INPUT FIELD"
          name="INPUT FIELD"
          placeholder="INPUT FIELD"
          disabled
        />
      </Form>
    )

    expect(await axe(container)).toHaveNoViolations()
  })

  it('renders a field in error state', async () => {
    const errorText = 'This field is required'

    const { container, findByText, getByRole } = render(
      <Form>
        <InputField
          label="INPUT FIELD"
          name="INPUT FIELD"
          validation={{ required: errorText }}
        />
        <button type="submit">Submit</button>
      </Form>
    )
    userEvent.click(getByRole('button'))
    await findByText(errorText)

    expect(container).toMatchSnapshot()
  })

  it('renders a field in error state - has no programmatically detectable a11y issues', async () => {
    const errorText = 'This field is required'

    const { container, findByText, getByRole } = render(
      <Form>
        <InputField
          label="INPUT FIELD"
          name="INPUT FIELD"
          validation={{ required: errorText }}
        />
        <button type="submit">Submit</button>
      </Form>
    )

    userEvent.click(getByRole('button'))
    await findByText(errorText)

    expect(await axe(container)).toHaveNoViolations()
  })
})
