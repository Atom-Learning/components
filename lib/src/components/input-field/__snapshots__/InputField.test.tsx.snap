// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField component renders a field in error state 1`] = `
@media  {
  .c-dhzjXW {
    display: flex;
  }

  .c-cVSpVL {
    color: var(--colors-tonal500);
    font-family: var(--fonts-body);
    margin: 0;
  }

  .c-cHbqOs {
    -webkit-appearance: none;
    appearance: none;
    border: 1px solid var(--colors-tonal400);
    border-radius: var(--radii-0);
    box-shadow: none;
    box-sizing: border-box;
    color: var(--colors-tonal600);
    cursor: text;
    display: block;
    font-family: var(--fonts-body);
    padding-left: var(--space-3);
    padding-right: var(--space-3);
    transition: all 100ms ease-out;
    width: 100%;
  }

  .c-cHbqOs:focus {
    border-color: var(--colors-primary);
    outline: none;
  }

  .c-cHbqOs[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }

  .c-cHbqOs::placeholder {
    color: var(--colors-tonal300);
    opacity: 1;
  }

  .c-fNCeYG {
    color: var(--colors-danger);
    margin-left: var(--space-1);
    font-weight: 400;
  }

  .c-hMRxhp {
    display: inline-block;
    fill: none;
    flex-shrink: 0;
    stroke: currentcolor;
    stroke-linecap: round;
    stroke-linejoin: round;
    vertical-align: middle;
  }

  .c-dyvMgW {
    font-family: var(--fonts-body);
    font-weight: 400;
    margin: 0;
  }

  .c-dyvMgW > .c-dyvMgW:before,
  .c-dyvMgW > .c-dyvMgW:after {
    display: none;
  }
}

@media  {
  .c-cVSpVL-gvmVBy-size-md {
    font-size: var(--fontSizes-md);
    line-height: 1.5;
  }

  .c-cVSpVL-gvmVBy-size-md::before {
    content: '';
    margin-bottom: -0.3864em;
    display: table;
  }

  .c-cVSpVL-gvmVBy-size-md::after {
    content: '';
    margin-top: -0.3864em;
    display: table;
  }

  .c-cVSpVL-grKRUr-type-block {
    display: block;
    font-weight: 600;
  }

  .c-cHbqOs-fwoCmE-size-md {
    height: var(--sizes-4);
    font-size: var(--fontSizes-md);
    line-height: 2;
  }

  .c-cHbqOs-bXVTjQ-state-error {
    border: 1px solid var(--colors-danger);
  }

  .c-PJLV-iymbh-theme-error {
    color: var(--colors-danger);
  }

  .c-hMRxhp-cyeZeh-size-sm {
    height: var(--sizes-1);
    width: var(--sizes-1);
    stroke-width: 1.5;
  }

  .c-dyvMgW-bndJoy-size-sm {
    font-size: var(--fontSizes-sm);
    line-height: 1.53;
  }

  .c-dyvMgW-bndJoy-size-sm::before {
    content: '';
    margin-bottom: -0.4056em;
    display: table;
  }

  .c-dyvMgW-bndJoy-size-sm::after {
    content: '';
    margin-top: -0.4056em;
    display: table;
  }
}

@media  {
  .c-PJLV-ikLAKdM-css {
    margin: auto;
    height: 100px;
    width: 100px;
  }

  .c-dhzjXW-iwHglP-css {
    justify-content: space-between;
    align-items: center;
    margin-bottom: var(--space-3);
  }

  .c-dhzjXW-ijkHUQU-css {
    margin-top: var(--space-2);
  }

  .c-hMRxhp-ijkHsGW-css {
    margin-right: var(--space-2);
  }

  .c-dyvMgW-iiTbKsx-css {
    transform: translateY(var(--space-0));
  }
}

<div>
  <form
    aria-label="form"
    class="c-PJLV"
  >
    <div
      class="c-PJLV"
    >
      <div
        class="c-dhzjXW c-dhzjXW-iwHglP-css"
      >
        <label
          class="c-cVSpVL c-cVSpVL-gvmVBy-size-md c-cVSpVL-grKRUr-type-block"
          for="INPUT FIELD"
        >
          INPUT FIELD
          <span
            aria-hidden="true"
            class="c-fNCeYG"
          >
            *
          </span>
        </label>
      </div>
      <input
        class="c-cHbqOs c-cHbqOs-fwoCmE-size-md c-cHbqOs-bXVTjQ-state-error"
        id="INPUT FIELD"
        name="INPUT FIELD"
        type="text"
      />
      <div
        class="c-dhzjXW c-PJLV c-PJLV-iymbh-theme-error c-dhzjXW-ijkHUQU-css"
      >
        <svg
          aria-hidden="true"
          class="c-hMRxhp c-hMRxhp-cyeZeh-size-sm c-hMRxhp-ijkHsGW-css"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 8v5"
          />
          <line
            x1="12"
            x2="12"
            y1="16"
            y2="16"
          />
          <circle
            cx="12"
            cy="12"
            r="10"
          />
        </svg>
        <span
          class="c-dyvMgW c-dyvMgW-bndJoy-size-sm c-dyvMgW-iiTbKsx-css"
        >
          This field is required
        </span>
      </div>
    </div>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`InputField component renders a field with a number input 1`] = `
@media  {
  .c-dhzjXW {
    display: flex;
  }

  .c-cVSpVL {
    color: var(--colors-tonal500);
    font-family: var(--fonts-body);
    margin: 0;
  }

  .c-cHbqOs {
    -webkit-appearance: none;
    appearance: none;
    border: 1px solid var(--colors-tonal400);
    border-radius: var(--radii-0);
    box-shadow: none;
    box-sizing: border-box;
    color: var(--colors-tonal600);
    cursor: text;
    display: block;
    font-family: var(--fonts-body);
    padding-left: var(--space-3);
    padding-right: var(--space-3);
    transition: all 100ms ease-out;
    width: 100%;
  }

  .c-cHbqOs:focus {
    border-color: var(--colors-primary);
    outline: none;
  }

  .c-cHbqOs[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }

  .c-cHbqOs::placeholder {
    color: var(--colors-tonal300);
    opacity: 1;
  }
}

@media  {
  .c-cVSpVL-gvmVBy-size-md {
    font-size: var(--fontSizes-md);
    line-height: 1.5;
  }

  .c-cVSpVL-gvmVBy-size-md::before {
    content: '';
    margin-bottom: -0.3864em;
    display: table;
  }

  .c-cVSpVL-gvmVBy-size-md::after {
    content: '';
    margin-top: -0.3864em;
    display: table;
  }

  .c-cVSpVL-grKRUr-type-block {
    display: block;
    font-weight: 600;
  }

  .c-cHbqOs-fwoCmE-size-md {
    height: var(--sizes-4);
    font-size: var(--fontSizes-md);
    line-height: 2;
  }
}

@media  {
  .c-PJLV-ikLAKdM-css {
    margin: auto;
    height: 100px;
    width: 100px;
  }

  .c-dhzjXW-iwHglP-css {
    justify-content: space-between;
    align-items: center;
    margin-bottom: var(--space-3);
  }
}

<div>
  <form
    aria-label="form"
    class="c-PJLV"
  >
    <div
      class="c-PJLV c-PJLV-ikLAKdM-css"
    >
      <div
        class="c-dhzjXW c-dhzjXW-iwHglP-css"
      >
        <label
          class="c-cVSpVL c-cVSpVL-gvmVBy-size-md c-cVSpVL-grKRUr-type-block"
          for="INPUT FIELD"
        >
          INPUT FIELD
        </label>
      </div>
      <input
        class="c-cHbqOs c-cHbqOs-fwoCmE-size-md"
        id="INPUT FIELD"
        inputmode="numeric"
        name="INPUT FIELD"
        pattern="[0-9]*"
        placeholder="001"
        type="text"
      />
    </div>
  </form>
</div>
`;

exports[`InputField component renders a field with a text input 1`] = `
@media  {
  .c-dhzjXW {
    display: flex;
  }

  .c-cVSpVL {
    color: var(--colors-tonal500);
    font-family: var(--fonts-body);
    margin: 0;
  }

  .c-cHbqOs {
    -webkit-appearance: none;
    appearance: none;
    border: 1px solid var(--colors-tonal400);
    border-radius: var(--radii-0);
    box-shadow: none;
    box-sizing: border-box;
    color: var(--colors-tonal600);
    cursor: text;
    display: block;
    font-family: var(--fonts-body);
    padding-left: var(--space-3);
    padding-right: var(--space-3);
    transition: all 100ms ease-out;
    width: 100%;
  }

  .c-cHbqOs:focus {
    border-color: var(--colors-primary);
    outline: none;
  }

  .c-cHbqOs[disabled] {
    opacity: 0.3;
    cursor: not-allowed;
  }

  .c-cHbqOs::placeholder {
    color: var(--colors-tonal300);
    opacity: 1;
  }
}

@media  {
  .c-cVSpVL-gvmVBy-size-md {
    font-size: var(--fontSizes-md);
    line-height: 1.5;
  }

  .c-cVSpVL-gvmVBy-size-md::before {
    content: '';
    margin-bottom: -0.3864em;
    display: table;
  }

  .c-cVSpVL-gvmVBy-size-md::after {
    content: '';
    margin-top: -0.3864em;
    display: table;
  }

  .c-cVSpVL-grKRUr-type-block {
    display: block;
    font-weight: 600;
  }

  .c-cHbqOs-fwoCmE-size-md {
    height: var(--sizes-4);
    font-size: var(--fontSizes-md);
    line-height: 2;
  }
}

@media  {
  .c-PJLV-ikLAKdM-css {
    margin: auto;
    height: 100px;
    width: 100px;
  }

  .c-dhzjXW-iwHglP-css {
    justify-content: space-between;
    align-items: center;
    margin-bottom: var(--space-3);
  }
}

<div>
  <form
    aria-label="form"
    class="c-PJLV"
  >
    <div
      class="c-PJLV c-PJLV-ikLAKdM-css"
    >
      <div
        class="c-dhzjXW c-dhzjXW-iwHglP-css"
      >
        <label
          class="c-cVSpVL c-cVSpVL-gvmVBy-size-md c-cVSpVL-grKRUr-type-block"
          for="INPUT FIELD"
        >
          INPUT FIELD
        </label>
      </div>
      <input
        class="c-cHbqOs c-cHbqOs-fwoCmE-size-md"
        id="INPUT FIELD"
        name="INPUT FIELD"
        placeholder="INPUT FIELD"
        type="text"
      />
    </div>
  </form>
</div>
`;
